2-0-data-structures-and-algorithms/src/test/java/com/bobocode/bst/BinarySearchTreeTest.java

```diff
@@ -1,82 +1,249 @@
 package com.bobocode.bst;
 
-import org.junit.jupiter.api.*;
-import org.junit.jupiter.params.*;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.function.Predicate;
 import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.*;
 import static org.junit.jupiter.params.provider.Arguments.arguments;
 
-
+@Slf4j
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 class BinarySearchTreeTest {
 
+    private static final Predicate<Field> SIZE_FIELD = field -> field.getName().equals("size") || field.getName().equals("length");
+    private static final Predicate<Field> NODE_FIELD = field -> field.getType().getSimpleName().equals("Node");
+    private static final Predicate<Field> ELEMENT_FIELD = field -> field.getName().equals("element")
+            || field.getName().equals("item")
+            || field.getName().equals("value");
+    private static final Predicate<Field> LEFT_FIELD = field -> field.getName().equals("left");
+    private static final Predicate<Field> RIGHT_FIELD = field -> field.getName().equals("right");
+
     private static final Integer[] someElements = {10, 9, 11, 8, 12, 7};
+    private BinarySearchTree<Integer> tree;
 
     @Test
     @Order(1)
-    void createWithElements() {
-        BinarySearchTree<Integer> bst = RecursiveBinarySearchTree.of(someElements);
-        for (var e: someElements) {
-            assertThat(bst.contains(e)).isTrue();
-        }
-        assertThat(bst.size()).isEqualTo(someElements.length);
+    void checkProperNodeClassName() {
+        tree = new RecursiveBinarySearchTree<>();
+        Class<?> innerClass = getInnerClass();
+        String name = innerClass.getSimpleName();
+
+        assertThat(name).isEqualTo("Node");
     }
 
     @Test
     @Order(2)
-    void insertUniqueElements() {
-        BinarySearchTree<Integer> bst = RecursiveBinarySearchTree.of();
-        for (int i = 0; i < someElements.length; i++) {
-            var e = someElements[i];
-            assertThat(bst.contains(e)).isFalse(); //does not contain
-            assertThat(bst.size()).isEqualTo(i);
+    void checkProperTreeFields() {
+        tree = new RecursiveBinarySearchTree<>();
```

Comment:
@erneman1 why don't you move initialization to the field declaration? 
`private BinarySearchTree<Integer> tree = new RecursiveBinarySearchTree<>();`
**JUnit creates a new instance for every test method**...